### ISSUE TITLE
Question for mapOfTHreadIds

### ISSUE THREAD

**Issue by chakku000** [onderkalaci/dyndatarace#1]
I'm looking your code of PureHappensBefore/PureHappensBeforeImp.cpp.
And I have a question for your code.

How works `mapOfThreadIds'?
When I read line82, it looks map of : PIN's ID → System ID:.
https://github.com/onderkalaci/dyndatarace/blob/master/sources/PureHappensBefore/PureHappensBeforeImp.cpp#L82
But I read 84,85 line, this code looks like use mapOfThreadIds as map of : System ID → PIN's ID:.
https://github.com/onderkalaci/dyndatarace/blob/master/sources/PureHappensBefore/PureHappensBeforeImp.cpp#L84-L85

Is this `mapOfThreadIds` is used as map of :PIN's ID <--> System ID:  Bidirectional transformation?

**Comment by onderkalaci** [comment]
I think it is map[ParentThreadId] = [List of Child ThreadIds] as declared [here](https://github.com/onderkalaci/dyndatarace/blob/master/sources/PureHappensBefore/PureHappensBeforeImp.cpp#L55).

Note that a child thread could be the parent of other threads. (I can see that it's hard to follow. I wish I had better coding conventions/variable names.)

**Comment by chakku000** [comment]
The type of mapThreadIds is map<THREADID,THREADID>, so I think mapOfThreadIds[ParentThradId] can't express [List of Child ThreadIds]. Because THREADID is not list type.

I think following.
PIN_GetParentTid() get System ID (not PIN'S ID), so in order to get Parent thread's PIN'S ID, there need transformation from System ID to PIN's ID. Is it right?
